nautobot_firewall_models/models/core_models.py

```diff
@@ -656,3 +656,227 @@ def to_json(self):
     def __str__(self):
         """Stringify instance."""
         return self.name
+
+
+@extras_features(
+    "custom_fields",
+    "custom_links",
+    "custom_validators",
+    "export_templates",
+    "graphql",
+    "relationships",
+    "statuses",
+    "webhooks",
+)
+class NATPolicyRule(PrimaryModel):
+    """
+    A NATPolicyRule is the equivalent of a single in a firewall policy or access list.
+
+    NAT policies are typically made up of several individual rules.
+    """
+
+    # Metadata
+    name = models.CharField(max_length=100)
+    tags = TaggableManager(through=TaggedItem)
+    mode = models.CharField(choices=choices.MODE_CHOICES, max_length=20)
+    log = models.BooleanField(default=False)
+    status = StatusField(
+        on_delete=models.PROTECT,
+        related_name="%(app_label)s_%(class)s_related",  # e.g. dcim_device_related
+        default=get_default_status,
+    )
+    request_id = models.CharField(max_length=100, null=True, blank=True)
+    description = models.CharField(max_length=200, null=True, blank=True)
+    index = models.PositiveSmallIntegerField(null=True, blank=True)
+
+    # Data that can not undergo a translation
+    source_users = models.ManyToManyField(to=UserObject, through="NATSrcUserM2M", related_name="nat_policy_rules")
```

Comment:
I'm not sure I understand why users are an option for a NAT. I've never seen users with NAT. It's always IP addresses and ports. Happy to be corrected though.

Reply:
I have removed source users. Multiple people had this thought and seeing and I have also looked at configuration for various vendors and couldn't find user as an option for NAT rules.